examples/integration_test.py

```diff
@@ -565,6 +565,175 @@ def my_on_revoke(consumer, partitions):
     c.close()
 
 
+def verify_batch_consumer():
+    """ Verify basic batch Consumer functionality """
+
+    # Consumer config
+    conf = {'bootstrap.servers': bootstrap_servers,
+            'group.id': 'test.py',
+            'session.timeout.ms': 6000,
+            'enable.auto.commit': False,
+            'api.version.request': api_version_request,
+            'on_commit': print_commit_result,
+            'error_cb': error_cb,
+            'default.topic.config': {
+                'auto.offset.reset': 'earliest'
+            }}
+
+    # Create consumer
+    c = confluent_kafka.Consumer(**conf)
+
+    # Subscribe to a list of topics
+    c.subscribe([topic])
+
+    max_msgcnt = 100
+    msgcnt = 0
+
+    while True:
+        # Consume until EOF or error
+
+        # Consume message (error()==0) or event (error()!=0)
+        msglist = c.consume(max_msgcnt, 1.0)
```

Comment:
Knowing that there is at least N messages in the topic (thanks to the produce tests), and where N > max_msgcnt (which it should be), we should check that we actually get max_msgcnt back for each call until we've hit our expected message count (which is some arbitrary value <=N).
Relying on EOF is not safe since the topic may feature multiple partitions and we don't know what the message distribution is like.